<?xml version="1.0"?>

<launch>
    <!-- static_transform_publisher x y z yaw pitch roll frame_id child_frame_id -->
    <!-- <node pkg="tf2_ros" type="static_transform_publisher" name="static_world_publisher" args="0.0 0.0 0.508978 0.0 0.0 0.0 world base_link" /> -->
    
    <!-- VIVE meshes -->
    <rosparam param="/vive_node/hmd_mesh_path">"package://vive_bridge/meshes/vive_pro_headset/vive_pro_headset.dae"</rosparam>
    <rosparam param="/vive_node/controller_mesh_path">"package://vive_bridge/meshes/vr_controller_vive_1_5/vr_controller_vive_1_5.dae"</rosparam>
    <rosparam param="/vive_node/tracker_mesh_path">"package://vive_bridge/meshes/TRACKER-3D.dae"</rosparam>
    <rosparam param="/vive_node/lighthouse_mesh_path">"package://vive_bridge/meshes/lh_basestation_vive/lh_basestation_vive.dae"</rosparam>

    <!-- Offset transformation parameters -->
    <rosparam param="/vive_node/inertial_frame">"root"</rosparam>
    <rosparam param="/vive_node/vr_frame">"world_vr"</rosparam>
    <!-- <rosparam param="/vive_node/vr_x_offset">-1.89223626887731</rosparam>
    <rosparam param="/vive_node/vr_y_offset">-2.608765797841659</rosparam>
    <rosparam param="/vive_node/vr_z_offset">2.9165620246814097</rosparam>
    <rosparam param="/vive_node/vr_yaw_offset">-0.607716823926031</rosparam>
    <rosparam param="/vive_node/vr_pitch_offset">0.01411116306491677</rosparam>
    <rosparam param="/vive_node/vr_roll_offset">1.571784220099083</rosparam> -->

    <!-- Launch node through steam with shell script -->
    <node name="vive_node" pkg="vive_bridge" type="launch.sh" output="screen" required="true"/>
</launch>